src/transform/ancestor.js

```diff
@@ -118,7 +118,7 @@ function canBeLifted(doc, range) {
   }
 }
 
-// :: (Node, Pos, ?Pos) → bool
+// :: (Node, Pos, ?Pos) → ?{found: any, range: any}
```

Comment:
I meant assigning to their prototypes. You can do that, but if you want to get any strong typing, you need to declare them when declaring the class, which would break your use-case of doing it in separate modules.

It's not about whether the loop terminates (Flow trusts you with that), it's about whether the function returns a value. If you don't end your function body with a return, it's pretty sure your function could return undefined, which is not the case in a few functions.